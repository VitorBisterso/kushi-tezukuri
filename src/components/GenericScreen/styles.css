.genericScreen-container {
  display: flex;
  flex: 1;
  flex-direction: column;
  margin: 0 5vw;
}

.genericScreen-content {
  display: flex;
  flex: 1;
  flex-direction: row;
  justify-content: space-between;
}

.genericScreen-title {
  font-family: 'Mali-Bold';
  font-size: var(--TEXT-BIGGEST);
  color: var(--COLOR-GOLD);
}

.genericScreen-content p {
  max-width: 50%;
  font-size: var(--TEXT-BIG);
  color: var(--COLOR-BLACK);
  text-align: justify;
}

.genericScreen-content img {
  margin-top: -5vw;
  width: 40%;
}

@media (max-width: 767px) {
  .genericScreen-container {
    margin: 0 2vw;
  }

  .genericScreen-content {
    flex-direction: column;
  }

  .genericScreen-title {
    font-size: var(--TEXT-VERY_BIG);
  }

  .genericScreen-content p {
    max-width: 80%;
    font-size: var(--TEXT-REGULAR);
  }

  .genericScreen-content img {
    margin: 2vw 0;
    width: 75%;
  }
}
