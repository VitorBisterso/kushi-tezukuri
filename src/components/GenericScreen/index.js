import React from 'react'
import PropTypes from 'prop-types'
import Fade from 'react-reveal/Fade'

import './styles.css'

const GenericScreen = ({ title, mainText, image }) => (
  <div className="genericScreen-container">
    <p className="genericScreen-title">{title}</p>
    <div className="genericScreen-content">
      <Fade left>
        <p>{mainText}</p>
      </Fade>
      <Fade right>
        <img src={image} alt="genericScreen-svg" />
      </Fade>
    </div>
  </div>
)

GenericScreen.propTypes = {
  title: PropTypes.string.isRequired,
  mainText: PropTypes.string.isRequired,
  image: PropTypes.node.isRequired,
}

export default GenericScreen
