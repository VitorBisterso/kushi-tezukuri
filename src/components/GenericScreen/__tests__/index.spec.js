import React from 'react'
import { shallow } from 'enzyme'

import GenericScreen from '../index'

import image from '../../../assets/img/mask.jpg'

describe('GenericScreen', () => {
  it('should render correctly', () => {
    const component = shallow(
      <GenericScreen title="Test" mainText="This is a test" image={image} />
    )

    expect(component).toMatchSnapshot()
  })
})
