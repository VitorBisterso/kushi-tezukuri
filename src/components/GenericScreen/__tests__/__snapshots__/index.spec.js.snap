// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericScreen should render correctly 1`] = `
<div
  className="genericScreen-container"
>
  <p
    className="genericScreen-title"
  >
    Test
  </p>
  <div
    className="genericScreen-content"
  >
    <Fade
      left={true}
    >
      <p>
        This is a test
      </p>
    </Fade>
    <Fade
      right={true}
    >
      <img
        alt="genericScreen-svg"
        src=""
      />
    </Fade>
  </div>
</div>
`;
